src/parser/ValueParser.ts

```diff
@@ -283,6 +293,39 @@ export class ValueParser {
     return valueType;
   }
 
+  private parseArraryReferenceNode(referenceNode: ts.TypeReferenceNode): ArrayType | null {
+    if (referenceNode.typeName.getText() !== 'Array') {
+      return null;
+    }
+    if (referenceNode.typeArguments === undefined || referenceNode.typeArguments.length !== 1) {
+      return null;
+    }
+
+    const elementType = this.valueTypeFromTypeNode(referenceNode.typeArguments[0]);
+
+    return {
+      kind: ValueTypeKind.arrayType,
+      elementType,
+    };
+  }
+
+  private parseRecordReferenceNode(referenceNode: ts.TypeReferenceNode): DictionaryType | null {
+    if (referenceNode.typeName.getText() !== 'Record') {
```

Comment:
@imWildCat do you think we also need to support `Map<type, type>`?

Reply:
I think so. Because `Map` is an ES6 native type: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#objects_vs._maps

However, we can do it in another PR.

Reply:
It's not much work. I will do it here.